src/main/java/com/github/joschi/jersey/security/smime/EnvelopedReader.java

```diff
@@ -61,7 +64,7 @@ public EnvelopedInput readFrom(Class<EnvelopedInput> type, Type genericType, Ann
         if (headers.containsKey("Content-Type")) {
             headerString.append("Content-Type: ").append(headers.getFirst("Content-Type")).append(CRLF);
         }
-        if (headers.containsKey("Content-Transfer-Encoding")) {
+        if ("base64".equalsIgnoreCase(headers.getFirst("Content-Transfer-Encoding"))) {
```

Comment:
I think it's better to always pass the "Content-Transfer-Encoding" header unconditionally, for two reasons:

* `MimeBodyPart` should know best which encodings it supports
* In case of an invalid/unknown encoding, with the check in place, it changes the exception to a rather generic one:
  > org.bouncycastle.cms.CMSException: Malformed content

  Without the check, it's pretty clear what is wrong:
  > javax.mail.MessagingException: can't extract input stream: java.io.IOException: Unknown encoding: some-unknown-encoding`.